kcqrs-client/src/main/kotlin/com/clouway/kcqrs/client/HttpEventStore.kt

```diff
@@ -0,0 +1,130 @@
+                val resp = response.parseAs(SaveEventsResponseDto::class.java)
+                return SaveEventsResponse.Success(aggregateId, resp.version)
+            }
+
+            if (response.statusCode == HttpStatusCodes.STATUS_CODE_CONFLICT) {
+                val resp = response.parseAs(SaveEventsResponseDto::class.java)
+                return SaveEventsResponse.EventCollision(aggregateId, resp.version)
+            }
+
+        } catch (ex: IOException) {
+            return SaveEventsResponse.ErrorInCommunication
+        }
+
+        return SaveEventsResponse.Error("Generic Error")
+    }
+
+    override fun getEvents(aggregateId: String): GetEventsResponse {
+        val request = requestFactory.buildGetRequest(GenericUrl(endpoint.toString() + "/v1/aggregates/$aggregateId"))
+        request.throwExceptionOnExecuteError = false
+        try {
+            val response = request.execute()
+
+            // Aggregate was not found and no events cannot be returned
+            if (response.statusCode == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
+                return GetEventsResponse.AggregateNotFound
+            }
+
+            if (response.isSuccessStatusCode) {
+                val resp = response.parseAs(GetEventsResponseDto::class.java)
+
+                val events = resp.events.map { EventPayload(it.kind, it.timestamp, it.identityId, Binary(it.payload)) }
+
+                return GetEventsResponse.Success(aggregateId, resp.aggregateType, null, resp.version, events)
+            }
+
+            return GetEventsResponse.Error("got unknown error")
+
+        } catch (ex: IOException) {
+            return GetEventsResponse.ErrorInCommunication
+        }
+    }
+
+    override fun revertLastEvents(aggregateId: String, count: Int): RevertEventsResponse {
+        val request = requestFactory.buildPatchRequest(
+                GenericUrl(endpoint.toString() + "/v1/aggregates/$aggregateId"),
+                JsonHttpContent(GsonFactory.getDefaultInstance(), RevertEventsRequestDto(aggregateId, count))
+        )
+        request.throwExceptionOnExecuteError = false
+        try {
+            val response = request.execute()
+
+            // Aggregate was not found, so no events cannot be returned
+            if (response.statusCode == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
+                return RevertEventsResponse.AggregateNotFound
+            }
+
+            if (response.isSuccessStatusCode) {
+                return RevertEventsResponse.Success
+            }
+
+            return RevertEventsResponse.Error("Generic Error")
+
+        } catch (ex: IOException) {
+            return RevertEventsResponse.Error("Communication Error")
+        }
+    }
+
+}
+
+internal data class GetEventsResponseDto(@Key @JvmField var aggregateId: String, @Key @JvmField var aggregateType: String, @Key @JvmField var version: Long, @Key @JvmField var topic: String, @Key @JvmField var events: List<EventPayloadDto>) : GenericJson() {
```

Comment:
I'm guessing you are using the ```@Key``` and ```@JvmField``` so that when the GenericJsone object is parsed we would be able to do something like ```json.get("aggregateId")``` ? Is it a requirement the fields to be var as well and not val ?

Reply:
Yes, it's required by the parser of the google-http-client library.